modules/vpc-endpoints/main.tf

```diff
@@ -24,7 +24,7 @@ resource "aws_vpc_endpoint" "this" {
   for_each = local.endpoints
 
   vpc_id            = var.vpc_id
-  service_name      = data.aws_vpc_endpoint_service.this[each.key].service_name
+  service_name      = try(each.value.service_endpoint_url, data.aws_vpc_endpoint_service.this[each.key].service_name)
```

Comment:
How `service_endpoint_url` should look for the FIPS environment?

Reply:
Sorry, I wasn't clear on this, I meant more so in the context of since you can override with inputs, users can use fips endpoints

see:
https://github.com/defenseunicorns/terraform-aws-vpc/pull/113/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR227

You'd probably able to do it with this module if we added another `data "aws_vpc_endpoint_service"` with different filters, but It doesn't seem like a super common enough request.

Reply:
I see. I think it should be named `service_endpoint` instead of `service_endpoint_url` to represent the actual type of the expected value. Could you change it and I will merge it? :)

Reply:
Sure, no problem.